### PR TITLE
MODORDSTOR-68 - Piece records should have format of the piece, not the poLine

### PR DESCRIPTION
**Pull request by khandramai**
[MODORDSTOR-68](https://issues.folio.org/browse/MODORDSTOR-68) - Piece records should have format of the piece, not the poLine

## Purpose
The receiving history has the poLine's order format, but for orders that aren't all physical or all electronic, it becomes ambiguous during receiving which piece you're actually receiving. In order to avoid this confusion we need to make changes to both piece (schema) and receiving history (schema and view).

## Approach
Schemas changed:
- Added **piece.format** - valid values are: "Physical", "Electronic", "Other"
- Added **receivingHistory.pieceFormat** -> **piece.format**
- Removed **receivingHistory.poLineOrderFormat**

## Pre-Merge Checklist
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [x] Examples exist for all schemas
  - [x] Descriptions exist for all schema properties
  - [x] All schemas pass raml-cop linting
- Does this introduce breaking changes?
  - [x] Were there any schema changes?
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
**mod-orders**
**mod-orders-storage**

- Do JIRAs exist to update the impacted modules?
[MODORDERS-194](https://issues.folio.org/browse/MODORDERS-194)
[MODORDSTOR-68](https://issues.folio.org/browse/MODORDSTOR-68) (mod-orders-storage, PR [#99](https://github.com/folio-org/mod-orders-storage/pull/99))

### CODE REVIEW

File: mod-orders-storage/examples/piece_collection.sample [hunk 1]
```diff
@@ -5,6 +5,7 @@
         "id": "5e317dc2-deeb-4429-b2a1-91e5cd0fd5f7",
         "caption": "Tutorial Volume 5",
         "comment": "Special Edition",
+        "format": "Electronic",
```
Comment (piotr-kalashuk) [inline mod-orders-storage/examples/piece_collection.sample]: Please add a few more pieces here of different format and content.

File: mod-orders-storage/schemas/piece.json [hunk 2]
```diff
@@ -16,6 +16,11 @@
       "description": "Free form commentary",
       "type": "string"
     },
+    "format": {
+      "description": "The format of the piece",
+      "type": "object",
```
Comment (piotr-kalashuk) [inline mod-orders-storage/schemas/piece.json]: Shouldn't it be of `string` type?

File: mod-orders-storage/examples/piece_collection.sample (hunk 1 above)
Comment (khandramai) [reply mod-orders-storage/examples/piece_collection.sample]: Done

File: mod-orders-storage/schemas/piece.json (hunk 2 above)
Comment (khandramai) [reply mod-orders-storage/schemas/piece.json]: Changed

File: mod-orders-storage/schemas/receiving_history.json [hunk 3]
```diff
@@ -50,16 +50,16 @@
...
-      "$ref": "order_format.json"
-    },
     "poLineReceiptStatus": {
       "description": "The purchase order line receipt status",
       "type": "object",
       "$ref": "receipt_status.json"
     },
+    "pieceFormat": {
+      "description": "The format of the piece",
+      "type": "object",
```
Comment (KVupp) [inline mod-orders-storage/schemas/receiving_history.json]: with this format as object and the piece.json as String, is the receiving_history_view able to map the values correctly?

Comment (khandramai) [reply mod-orders-storage/schemas/receiving_history.json]: Noted. Thanks.

**Pull request merged**